tests/cross/src/test/scala/tests/pc/CompletionErrorSuite.scala

```diff
@@ -0,0 +1,43 @@
+package tests.pc
+
+import tests.BaseCompletionSuite
+
+class CompletionErrorSuite extends BaseCompletionSuite {
+
+  check(
+    "issue-3421",
+    """|object Main{
+       |  def thing() = thi@@
+       |  def run() = {
+       |    val thing = ???
+       |  }
+       |}
+       |""".stripMargin,
+    "thing(): Any",
+    topLines = Some(1),
+    compat = Map(
+      "3" -> "thing: Any"
+    )
+  )
+
+  check(
+    "issue-3421-match",
+    """|
+       |trait Dependency
+       |object InvalidDependency extends Dependency
+       |object Main{
+       |  def exists(dep: Dependency) = {
+       |    deps match {
+       |      case invalid @ InvalidDependency => println(inv@@)
+       |    }
+       |  }
+       |}
+       |""".stripMargin,
+    "invalid: InvalidDependency.type",
```

Comment:
I tried to follow how we handle things in Scala 2, but in this case the Scala 2 compiler gives us a bit more info.